src/core/meshcop/tcat_agent.cpp

```diff
@@ -0,0 +1,524 @@
+{
+    static constexpr int kGeneralTlvs            = 0x1F;
+    static constexpr int kCommissioningTlvs      = 0x3F;
+    static constexpr int kExtractionTlvs         = 0x5F;
+    static constexpr int kTlvDecommissioningTlvs = 0x7F;
+    static constexpr int kApplicationTlvs        = 0x9F;
+
+    if (aTlvType <= kGeneralTlvs)
+    {
+        return kGeneral;
+    }
+    else if (aTlvType <= kCommissioningTlvs)
+    {
+        return kCommissioning;
+    }
+    else if (aTlvType <= kExtractionTlvs)
+    {
+        return kExtraction;
+    }
+    else if (aTlvType <= kTlvDecommissioningTlvs)
+    {
+        return kTlvDecommissioning;
+    }
+    else if (aTlvType <= kApplicationTlvs)
+    {
+        return kApplication;
+    }
+    else
+    {
+        return kInvalid;
+    }
+}
+
+bool TcatAgent::CanProcessTlv(uint8_t aTlvType) const
+{
+    CommandClass tlvCommandClass = GetCommandClass(aTlvType);
+    return IsCommandClassAuthorized(tlvCommandClass);
+}
+
+Error TcatAgent::HandleSingleTlv(const Message &aIncommingMessage, Message &aOutgoingMessage)
+{
+    Error    error = kErrorParse;
+    ot::Tlv  tlv;
+    uint16_t offset = aIncommingMessage.GetOffset();
+    uint16_t length;
+    bool     response = false;
+
+    VerifyOrExit(IsConnected(), error = kErrorInvalidState);
+    SuccessOrExit(error = aIncommingMessage.Read(offset, tlv));
+
+    if (tlv.IsExtended())
+    {
+        ot::ExtendedTlv extTlv;
+        SuccessOrExit(error = aIncommingMessage.Read(offset, extTlv));
+        length = extTlv.GetLength();
+        offset += sizeof(ot::ExtendedTlv);
+    }
+    else
+    {
+        length = tlv.GetLength();
+        offset += sizeof(ot::Tlv);
+    }
+
+    if (!CanProcessTlv(tlv.GetType()))
+    {
+        error = kErrorRejected;
+    }
+    else
+    {
+        switch (tlv.GetType())
+        {
+        case kTlvDisconnect:
+            error = kErrorAbort;
+            break;
+
+        case kTlvSetActiveOperationalDataset:
+            error = HandleSetActiveOperationalDataset(aIncommingMessage, offset, length);
+            break;
+
+        case kTlvStartThreadInterface:
+            error = HandleStartThreadInterface();
+            break;
+
+        case kTlvStopThreadInterface:
+            Get<Mle::MleRouter>().Stop();
+            error = kErrorNone;
+            break;
+
+        case kTlvSendApplicationData:
+            LogInfo("Application data len:%d, offset:%d", length, offset);
+            mAppDataReceiveCallback.InvokeIfSet(&GetInstance(), &aIncommingMessage, offset,
+                                                MapEnum(mCurrentApplicationLayer), mCurrentServiceName);
+            response = true;
+            error    = kErrorNone;
+            break;
+
+        default:
+            error = kErrorInvalidCommand;
+        }
+    }
+
+    if (!response)
+    {
+        ot::Tlv    statusTlv;
+        StatusCode statusCode;
+
+        statusTlv.SetType(kTlvResponseWithStatus);
```

Comment:
`statusTlv` is defined, configured here, but not used after that. Remove `statusTlv` ?

Reply:
fixed.